storage/shard.go

```diff
@@ -352,6 +352,26 @@ func (s *dbShard) OnRetrieveBlock(
 	})
 }
 
+func (s *dbShard) OnEvictedFromWiredList(id ident.ID, blockStart time.Time) {
+	s.RLock()
+	entry, _, err := s.lookupEntryWithLock(id)
+	s.RUnlock()
+
+	if err != nil && err != errShardEntryNotFound {
+		return // Shard is probably closing
+	}
+
+	if entry == nil {
+		// This could happen in a situation where the WiredList had a reference
```

Comment:
Not sure I follow. I thought Tick's couldn't flush retrieved blocks?

Reply:
I'm not sure if when I wrote this I meant flushed as in written to a fileset file, or I meant flushed as in evicted.

Regardless I think you're right that this shouldn't happen. If a block hasn't been flushed to disk yet then it shouldn't be in the WiredList. If it has been flushed and its in the wired list because it was retrieved from disk, then the series shouldn't be evicted from memory because it has a block.

It still seems like something we should defend against. Maybe I should emit a warning log?

Reply:
replaced with an error log, will throw this up on the test cluster to test our theory that this doesn't happen